pkg/proxy/endpoints.go

```diff
@@ -102,6 +102,21 @@ type EndpointChangeTracker struct {
 	lastChangeTriggerTimes map[types.NamespacedName][]time.Time
 }
 
+// getChangesAndMarkApplied returns a list of pending endpointsChanges and marks
+// them as applied.
+func (ect *EndpointChangeTracker) getChangesAndMarkApplied() []*endpointsChange {
```

Comment:
add a unit test for this?

Reply:
may need a more significant name than `getChangesAndMarkApplied`. This function essentailly emits the changes in cache and refresh the cache. Plus, maybe other stuff

Naming Idea:
`CheckoutChanges`

Reply:
Thanks, `CheckoutChanges` is a much better name. Added some unit tests for this as well.